org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/matching/MatchLocator.java

```diff
@@ -3457,4 +3444,51 @@ protected boolean typeInHierarchy(ReferenceBinding binding) {
 	}
 	return false;
 }
+/*
+ * Go through the nest hierarchy of a type, to try to create a method handle for a method declaration.
+ * Does not try to create a method handle for the specified type, only for types higher in the nest hierarchy.
+ *
+ * See:
+ * https://github.com/eclipse-jdt/eclipse.jdt.core/issues/375
+ * https://bugs.eclipse.org/bugs/show_bug.cgi?id=468127
+ */
+private IJavaElement createHandleFromNestAncestors(AbstractMethodDeclaration method, IJavaElement type) {
```

Comment:
I don't know what to call this method, since its tailored to work only for the 1 current caller. We could maybe change the caller part for this method to make more sense.

What it does right now is, for binary (i.e. in a jar) nested types, go up the files and try to find the specified method declaration.

So if e.g. ```MyType$2$3``` is passed, the method will look for the methods in:

1. ```MyType$2```
2. ```MyType```

Its very lengthy and nested due to the many checks needed to know have a binary type. But that can be shortened, if we agree on this as a solution.

@iloveeclipse  WDYT? I think this is an OK compromise between keeping the fix for https://bugs.eclipse.org/bugs/show_bug.cgi?id=468127 but not breaking the search/call hierarchy for methods called by anonymous types.

It only needs some code cleaning up from my POV. And I hope it doesn't introduce bugs...

Reply:
I think this approach is OK.